main/src/cgeo/geocaching/storage/DataStore.java

```diff
@@ -4360,7 +4360,7 @@ private static StringBuilder whereGeocodeIn(final Collection<String> geocodes) {
         }
         query.append(" FROM ").append(dbTableWaypoints).append(", ").append(dbTableCaches).append(" WHERE ").append(dbTableWaypoints)
                 .append(".geocode == ").append(dbTableCaches).append(".geocode AND ").append(where)
-                .append(" LIMIT " + (Settings.getKeyInt(R.integer.waypoint_threshold_max) * 2));  // Hardcoded limit to avoid memory overflow
+                .append(" LIMIT " + (Math.max(10, Settings.getKeyInt(R.integer.waypoint_threshold_max)) * 2));  // Hardcoded limit to avoid memory overflow
```

Comment:
Would that mean, that no more than 10 waypoints will be shown on the map ever again?
I know of multi caches which have 40 or 50 waypoints for a single cache (see GC24CQN for example), those would become unplayable, wouldn't they?

Reply:
No, it just means that the limit will.never be less than 10, regardless of the setting. I put this guard to prevent the theoretical case that the setting returns e.g. a negative number

Reply:
Oops, my bad, I've read `min` somehow... Sorry!